dash-renderer/package.json

```diff
@@ -22,6 +21,34 @@
   "author": "chriddyp",
   "license": "MIT",
   "dependencies": {
+    "react": "16.8.6",
+    "react-dom": "16.8.6",
+    "prop-types": "15.7.2",
+    "check-prop-types": "1.1.2"
```

Comment:
Hmm, this is not what I thought the `dependencies` / `devDependencies` distinction meant. I thought basically if you reference a module via `import` / `require` in your source code it should be included in `dependencies`, because anyone else who imports your package in their own source will need to have those available. `devDependencies` then is just for things needed in your own build pipeline or tests.

That would be a much smaller change than we have here. Is that not the right way to think about this?

Reply:
this is a change upon @Marc-Andre-Rivet 's comments, you might miss the context as I resolved the conversation. but open to change it back if the suggestion doesn't make sense

Reply:
I saw that comment, but either it's not correct or I'm misunderstanding what these fields are about. Let's figure out which it is 😄

Reply:
I haven't found the official documentation particularly clear on this, but here's a post that aligns with my understanding of deps vs devDeps: https://stackoverflow.com/a/22004559/9188800

The key, to my mind, is the part about transitivity:

> - `dependencies` are installed transitively: if A requires B, and B requires C, then C gets installed, otherwise, B could not work, and neither would A.
> - `devDependencies` is not installed transitively. E.g. we don't need to test B to test A, so B's testing dependencies can be left out.

So for our own uses it doesn't matter what's in `dependencies` vs `devDependencies`. The distinction only matters when someone else includes `dash-renderer` as a dependency in their own `package.json` - in that case they'll only get `dependencies`, and they'll expect that this will let them resolve all the imports in our source code.

Reply:
@byronz After talking with @alexcjohnson about this and 🍜 , let's keep these changes.

Additionally, let's change the entrypoint so that it points to the actual build artifact.
https://github.com/plotly/dash/pull/874/files/47ee2b07449f4ea5736d3dbc0906a620c5f261a9..d273bdb480c0779cefe55e3855e1f2d1b93e309f#diff-af8bab870558d6680886cec3032ab37cR5

Should become `dash_renderer/dash_renderer.min.js` or the likes.

Reply:
wait, if we're going that way, `check-prop-types` shouldn't be in `dependencies` either